src/Optimization/UI/BlinkUtils.jl

```diff
@@ -0,0 +1,352 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+module BlinkUtils
+
+# using Blink, WebIO, Interact, Interact.CSSUtil, DataFrames, CSV, TableView, Widgets
+using Blink, WebIO
+
+
+
+# function WebIO.render(x::HTML{String}) 
+#     # @info "--------------------------- $x"
+#     # return dom"div"(; setInnerHtml=WebIO.richest_html(x))
+#     return Node(:span ; setInnerHtml=WebIO.richest_html(x))
+# end
+
+# function WebIO.richest_html(val)
+#     # @info "RG $val"
+#     topmime = WebIO.richest_mime(val)
+#     str_repr = WebIO.stringmime(topmime, val)
+#     if topmime == MIME("text/html")
+#         # @info "1 $str_repr"
+#         return str_repr # |> WebIO.encode_scripts
+#     elseif topmime in map(MIME, ["image/png", "image/jpeg"])
+#         @info "2"
+#         return "<img src='data:image/png;base64,$str_repr'></img>"
+#     elseif topmime == MIME("image/svg+xml")
+#         @info "3"
+#         return str_repr
+#     else
+#         @info "4 $str_repr" 
+#         return str_repr # |> WebIO.encode_scripts
+#         # return "<pre>$str_repr</pre>"
+#     end
+# end
+
+
+struct HTMLParser
+    _original_text
+    _text
+    _variables
+end
+
+function HTMLParser(filename::String; kwargs...)
+    sl = readlines(filename)
+    return BlinkUtils.HTMLParser(sl; kwargs...)
+end
+
+function HTMLParser(text::AbstractArray{String}; replace_in_html = nothing, kwargs...)
```

Comment:
I'm not keen on the idea of pre-parsing html files before passing them to Blink. It seems to add an unnecessary layer of complexity to the development.

Are relative paths not workable (the logos actually aren't even displaying on my machine)? Are these comment-based hooks necessary when you can (presumably, correct me if I'm wrong) use queryselector?

If we take this idea further, we'll almost be developing our own HTML Julia wrapper.